secure.go

```diff
@@ -101,6 +128,11 @@ func (s *Secure) SetBadHostHandler(handler http.Handler) {
 // Handler implements the http.HandlerFunc for integration with the standard net/http lib.
 func (s *Secure) Handler(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if s.opt.nonceEnabled {
+			ctx := context.WithValue(r.Context(), nonceKey, randNonce(rand.NewSource(time.Now().UnixNano()), s.opt.CSPNonceByteAmount))
```

Comment:
Can we expose a WithCSPNonce function which takes a context.Context and a nonce string, and performs this WithValue internally.

One use-case is explained in a comment below (but tl;dr: if the HTTP handler makes a bunch of RPCs, I'd like to propagate the nonce through these RPCs so all the sub-handlers can use it to construct their individual HTML components).

Another is for testing. To test that my HTTP handler is propagating the nonce correctly, I need some way of setting it to a predictable value; I suppose I could use the whole handler wrapper mechanism to test, but that seems like overkill.